### PR TITLE
[Internal] Category: Fixes Null check for clients sending Content-Type without message body

### PR DESCRIPTION
**Pull request by DmitriMelnikov**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Include samples if adding new API, and include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/GatewayStoreClient.cs [hunk 1]
```diff
@@ -303,7 +303,10 @@ private static async Task<Stream> BufferContentIfAvailableAsync(HttpResponseMess
                     {
                         if (key.Equals(HttpConstants.HttpHeaders.ContentType, StringComparison.OrdinalIgnoreCase))
                         {
-                            requestMessage.Content.Headers.ContentType = new MediaTypeHeaderValue(request.Headers[key]);
+                            if (requestMessage.Content != null)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/GatewayStoreClient.cs:306]: Add tests

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/GatewayStoreClient.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/GatewayStoreClient.cs:306]: Please merge after this is addressed

**Comment by kirankumarkolli** [comment]
Closing in-active PR's

**Pull request closed without merging**